libexec/scoop-virustotal.ps1

```diff
@@ -104,45 +128,75 @@ Function SubmitMaybe-ToVirusTotal ($url, $app, $do_scan) {
         try {
             # Follow redirections (for e.g. sourceforge URLs) because
             # VirusTotal analyzes only "direct" download links
+            $url = $url.Split("#").GetValue(0)
             $new_redir = $url
             do {
                 $orig_redir = $new_redir
                 $new_redir = Get-RedirectedUrl $orig_redir
             } while ($orig_redir -ne $new_redir)
-            Invoke-RestMethod -Method POST -Uri "https://www.virustotal.com/ui/urls?url=$new_redir" | Out-Null
+            $uri = "https://www.virustotal.com/ui/urls?url=$new_redir"
+            $api_key = get_config("virustotal_api_key")
+            if ($api_key) {
+                $url += '&apikey=' + $api_key
+            }
+            Invoke-RestMethod -Method POST -Uri $uri | Out-Null
             $submitted = $True
         } catch [Exception] {
+            Warn("$app`: VirusTotal submission failed`: $($_.Exception.Message)")
             $submitted = $False
+            return
         }
     }
     else {
         $submitted = $False
     }
     if ($submitted) {
-        write-host -f darkred "$app`: $url unknown but submitted to VirusTotal"
+        Warn("$app`: not found`: submitted $url")
     }
     else {
-        write-host -f darkred "$app`: unknown, download $url & submit to https`://www.virustotal.com/"
+        Warn("$app`: not found`: manually submit $url")
     }
 }
 
 if(!$apps) {
     my_usage; exit 1
 }
 
+if($global -and !(is_admin)) {
```

Comment:
`--global` option isn't necessary because it only uses the manifest files. It doesn't check the installed program itself.

Reply:
Gotcha. I'll remove.